internal/datasourcev2/streaming_data_source.go

```diff
@@ -449,25 +435,17 @@ func (sp *StreamProcessor) GetFilterKey() string {
 	return sp.cfg.FilterKey
 }
 
-func processChangeset(changeSet changeSet) ([]any, error) {
-	if changeSet.intent == nil || changeSet.intent.Payloads[0].Code != "xfer-full" {
-		return convertChangesetEventsToPatchData(changeSet.events)
-	}
-
-	return convertChangesetEventsToPutData(changeSet.events)
-}
-
-func convertChangesetEventsToPatchData(events []es.Event) ([]any, error) {
-	updates := make([]interface{}, 0, len(events))
+func deserializeEvents(events []es.Event) ([]fdv2proto.Event, error) {
+	updates := make([]fdv2proto.Event, 0, len(events))
 
 	parseItem := func(r jreader.Reader, kind datakinds.DataKindInternal) (ldstoretypes.ItemDescriptor, error) {
 		item, err := kind.DeserializeFromJSONReader(&r)
 		return item, err
 	}
 
 	for _, event := range events {
-		switch event.Event() {
-		case putEventName:
+		switch fdv2proto.EventName(event.Event()) {
```

Comment:
I can't comment on the actual line, but `parseItem` here suffers from the same issue I pointed out above.

Reply:
It turns out this whole bit of logic (the event parsing) doesn't work properly due to the streaming JSON parser. I'm going to have to refactor this to more closely match how the existing datasource (V1) parsers work - which is two passes. 

The reason is because we might visit the "object" field _before_ we've visited the kind field.